### PR TITLE
Add special-token support to ordered and shuffled loaders

### PR DESCRIPTION
**Pull request by samuelstevens**
## Summary
- add a spec for CLS-only dataloader support
- add `tokens="special"` support to ordered and shuffled loaders for fixed integer layers
- fix special-token shard indexing across shard boundaries and add regression tests
- reject unsupported `OrderedDataLoader(..., layer="all")` configs up front

## Testing
- `uv run python -m pytest tests/test_ordered_dataloader.py -k 'special_tokens or constructor_rejects_layer_all' -q`
- `uv run python -m pytest tests/test_shuffled_dataloader.py -k 'special_tokens' -q`
- `uv run python -m pytest tests/test_shards_math.py -k 'special_tokens_with_cls_token' -q`
- `uvx ruff format --preview src/saev/data/ordered.py src/saev/data/shards.py src/saev/data/shuffled.py tests/test_ordered_dataloader.py tests/test_shuffled_dataloader.py tests/test_shards_math.py`
- `uvx ruff check --fix src/saev/data/ordered.py src/saev/data/shards.py src/saev/data/shuffled.py tests/test_ordered_dataloader.py tests/test_shuffled_dataloader.py tests/test_shards_math.py`

### CODE REVIEW

**Comment by samuelstevens** [comment]
@codex review please!

**Comment by samuelstevens** [comment]
@codex Review this PR.
